src/managers/SXCursorManager.hpp

```diff
@@ -0,0 +1,205 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <array>
+#include <cstdint>
+#include <hyprutils/math/Vector2D.hpp>
+#include "helpers/memory/Memory.hpp"
+
+extern "C" {
+#include <X11/Xcursor/Xcursor.h>
+}
+
+// gangsta bootleg XCursor impl. adidas balkanized
+struct SXCursorImage {
+    Vector2D              size;
+    Vector2D              hotspot;
+    std::vector<uint32_t> pixels; // XPixel is a u32
+    uint32_t              delay;  /* animation delay to next frame (ms) */
+};
+
+struct SXCursors {
+    std::vector<SXCursorImage> images;
+    std::string                shape;
+};
+
+// Taken from https://gitlab.freedesktop.org/xorg/lib/libxcursor/-/blob/master/src/library.c
+// clang-format off
+static std::array<const char*, 77> XCURSOR_STANDARD_NAMES = {
```

Comment:
why is this in the header

Reply:
moved it back to the .cpp file, had it mostly in the header to not having to scroll those extra 100 lines when implenting heh.